test/integration/hapi-statsd.tests.js

```diff
@@ -33,36 +36,94 @@ beforeEach(function(done) {
 		reply(new Error());
 	};
 
-	server.route({ method: ['GET','OPTIONS'], path: '/', handler: get, config: {cors: true}});
+	server.route({ method: ['GET','OPTIONS'], path: '/', handler: get, config: {cors: true} });
 	server.route({ method: 'GET', path: '/err', handler: err, config: {cors: true} });
-	server.route({ method: 'GET', path: '/test/{param}', handler: get, config: {cors: true}});
+	server.route({ method: 'GET', path: '/test/{param}', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/default', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/override', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/rename', handler: get, config: {cors: true} });
+	server.route({ method: 'GET', path: '/match/id', handler: get, config: {id: 'match-my-id', cors: true} });
 
 	server.register({
 		register: plugin,
-		options: { statsdClient: mockStatsdClient }
+		options: {
+			statsdClient: mockStatsdClient,
+			defaultFilter: {
+				enableCounter: false,
+				enableTimer: true,
+			},
+			filters: [
+				{ path: '/', enableCounter: true },
+				{ path: '/err', enableCounter: true },
+				{ path: '/test/{param}', enableCounter: true },
+				{ path: '/override', enableCounter: true, enableTimer: false },
+				{ path: '/rename', name: 'rename_stat', enableCounter: true, enableTimer: true },
+				{ id: 'match-my-id', name: 'match_id_stat', enableCounter: true, enableTimer: true },
+			],
+		},
 	}, done);
 });
 
 describe('hapi-statsd plugin tests', function() {
 
 	it('should expose statsd client to the hapi server', function() {
-
 		assert.equal(server.statsd, mockStatsdClient);
 	});
 
 	it('should report stats with no path in stat name', function(done) {
-
-		server.inject('/', function (res) {
+		server.inject('/', function(res) {
 			assert(mockStatsdClient.incStat == 'GET.200');
 			assert(mockStatsdClient.timingStat == 'GET.200');
 			assert(mockStatsdClient.timingDate instanceof Date);
 			done();
 		});
 	});
 
-	it('should report stats with path in stat name', function(done) {
+	it('should honor default filter', function(done) {
+		server.inject('/default', function(res) {
+			assert(mockStatsdClient.incStat == '');
+			assert(mockStatsdClient.timingStat == 'default.GET.200');
+			assert(mockStatsdClient.timingDate instanceof Date);
+			done();
+		});
+	});
+
+	it('should honor filter override', function(done) {
+		server.inject('/override', function(res) {
+			assert(mockStatsdClient.incStat == 'override.GET.200');
+			assert(mockStatsdClient.timingStat == '');
+			done();
+		});
+	});
 
-		server.inject('/test/123', function (res) {
+	it('should use cached value', function(done) {
```

Comment:
I guess one question to ask: Is this caching mechanism necessary? Or are we ok iterating through that array of filters on every request?